### PR TITLE
Republish drafts of other translations on PUT of new translation

### PR DESCRIPTION
**Pull request by brucebolt**
When a draft translation is added and another translation of the same content ID exists, we need to send the existing translation(s) downstream so their `available_translations` links can be updated to include the new translation.

We're not currently doing this, which leads to a situation where the link to the new translation never appears on the draft of the existing translations.

[Trello card](https://trello.com/c/8St6Zw4w)

### CODE REVIEW

File: app/commands/v2/put_content.rb [hunk 1]
```diff
@@ -200,6 +202,29 @@ def clear_draft_item_of_different_locale_but_matching_base_path
         )
       end
 
+      def update_draft_items_of_different_locale_but_matching_content_id
+        return unless payload[:base_path]
+        return if Document.where(content_id: payload[:content_id], locale: payload[:locale]).any?
```
Comment (Gweaton) [inline app/commands/v2/put_content.rb:207]: Is it worth adding a couple of tests for these early returns? Might also help serve as documentation.

**Review by Gweaton — APPROVED** [review]
I think this makes sense, just have a couple of qs/comments

File: app/commands/v2/put_content.rb [hunk 2]
```diff
@@ -200,6 +202,29 @@ def clear_draft_item_of_different_locale_but_matching_base_path
...
+
+        draft_editions_for_different_locale = Document
+          .where(content_id: payload[:content_id])
+          .where
+          .not(locale: payload[:locale])
+          .map(&:draft)
+          .compact
+
+        after_transaction_commit do
```
Comment (Gweaton) [inline app/commands/v2/put_content.rb:216]: What's the transaction here? Is it that the rest of the `call` method executes successfully and we only want to run this if it does? Or is it scoped to this method?

File: app/commands/v2/put_content.rb (hunk 1 above)
Comment (brucebolt) [reply app/commands/v2/put_content.rb:207]: Yes, that's a good idea, I'll add them.

File: app/commands/v2/put_content.rb (hunk 2 above)
Comment (brucebolt) [reply app/commands/v2/put_content.rb:216]: Yes, it's after the `call` (I think it's [this line](https://github.com/alphagov/publishing-api/blob/5ad088fe246c3dcaa4f73efdb9e637df6b35eaed/app/commands/base_command.rb#L21) that executes the block), so we only send the payload to content store once the data has actually been committed to the database.

**Pull request merged**